### PR TITLE
Cumulant spectral

### PR DESCRIPTION
**Pull request by yaoluo**
All the comments from last PR are fixed. Thank you, Ivan!

### CODE REVIEW

File: tests/test_spectral_cum.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    Returns
+    -------
+    phdisp : ppy.PhdispCalcMode
+
+    """
+    yml_path = os.path.join("refs", "sto_spectral-cum.yml")
+    spectral_path = os.path.join("refs", "sto_spectral_cumulant.h5")
+    return ppy.SpectralCumulant.from_hdf5_yaml(spectral_path, yml_path)
+
```
Comment (imaliyov) [inline tests/test_spectral_cum.py:21]: Please add just one more test that tests not the plot, but the SpectralCumulant object itself.

You could test just the shapes of the arrays, e.g.

```
    np.testing.assert_equal(obj.array1.shape, (181, 110))
    np.testing.assert_equal(obj.array2.shape, (181, 21))
```

**Review by imaliyov — COMMENTED** [review]
Thank you, Yao, this is a lot better!

Next time, please do not close the initial pull request, but continue commiting there. In this case, it will be easier to track changes.

Unless you had to change a lot...

Please add just one test that I specified in the file and we'll be good to go!

File: src/perturbopy/postproc/calc_modes/spectral_cumulant.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+        return cls(spectral_file, yaml_dict)
+
+    def plot_Aw(self, plt_loc, ax, ik=0, it=0, ib=0):
+        """
+        Plots the spectral function A(ω) for a given k-point, temperature, and band.
+
+        Parameters
+        ----------
+        plt_loc : matplotlib.pyplot
```
Comment (hurricane642) [inline src/perturbopy/postproc/calc_modes/spectral_cumulant.py:82]: Why do we need to pass the module to the function?

File: src/perturbopy/postproc/calc_modes/spectral_cumulant.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    temp_array : numpy.ndarray
+        Array of temperatures corresponding to the spectral data.
+    freq_array : numpy.ndarray
+        Array of energy values (ω) in electron volts (eV).
+    freq_step : float
+        Energy step size for the ω grid in eV.
+    Akw : numpy.ndarray
+        Spectral function data, indexed by k-point, band, temperature, and ω.
+    """
+    def __init__(self, spectral_file, pert_dict):
```
Comment (hurricane642) [inline src/perturbopy/postproc/calc_modes/spectral_cumulant.py:29]: Need docs here  for the class constructor

**Review by hurricane642 — CHANGES_REQUESTED** [review]
Overall looks good, thank you!
My comments mainly on the documentation of the functions

File: tests/test_spectral_cum.py (hunk 1 above)
Comment (yaoluo) [reply tests/test_spectral_cum.py:21]: do you have an example in the current perturbopy code?

File: src/perturbopy/postproc/calc_modes/calc_mode.py [hunk 4]
```diff
@@ -47,6 +47,7 @@ def __init__(self, pert_dict):
 
         # Extract calculation mode name from pert_dict
         self.calc_mode = pert_dict['input parameters']['after conversion'].pop('calc_mode')
+        self.prefix = pert_dict['input parameters']['after conversion'].pop('prefix')
```
[the remark this answers is not in the document]
Comment (hurricane642) [reply src/perturbopy/postproc/calc_modes/calc_mode.py:50]: Why is it resolved - we still need to fix this comment:)

File: src/perturbopy/postproc/calc_modes/spectral_cumulant.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    ----------
+    prefix : str
+        Prefix for the HDF5 file containing spectral function data.
+
+    Attributes
+    ----------
+    temp_array : numpy.ndarray
+        Array of temperatures corresponding to the spectral data.
+    freq_array : numpy.ndarray
+        Array of energy values (ω) in electron volts (eV).
```
[the remark this answers is not in the document]
Comment (hurricane642) [reply src/perturbopy/postproc/calc_modes/spectral_cumulant.py:23]: So, @yaoluo @imaliyov , what do you think on this?

**Review by hurricane642 — CHANGES_REQUESTED** [review]
There still things to rework, I've opened issue back

**Comment by yaoluo** [comment]
> I've opened issue back, see the comments earlier

I click the resolve when I changed the code. Now it is updated.

**Review by hurricane642 — APPROVED** [review]
Cool, LGTM! Thanks!

File: tests/test_spectral_cum.py (hunk 1 above)
Comment (hurricane642) [reply tests/test_spectral_cum.py:21]: Hey @yaoluo , I did't find the examples in the current perturbopy tests, but, if I un understand right, @imaliyov suggest the following idea: make class instance and then compare the size of the attributes with the reference one (or the values of this attributes with the reference one).

**Review by hurricane642 — APPROVED** [review]

**Pull request merged**